packages/contracts-bedrock/src/EAS/eip1271/EIP1271Verifier.sol

```diff
@@ -98,26 +101,42 @@ abstract contract EIP712Verifier is EIP712 {
                 )
             )
         );
-
-        if (ECDSA.recover(digest, signature.v, signature.r, signature.s) != request.attester) {
-            revert InvalidSignature();
-        }
+        _verifySignature(digest, signature, request.attester);
     }
 
     /// @notice Verifies delegated revocation request.
     /// @param request The arguments of the delegated revocation request.
     function _verifyRevoke(DelegatedRevocationRequest memory request) internal {
         RevocationRequestData memory data = request.data;
-        EIP712Signature memory signature = request.signature;
+        Signature memory signature = request.signature;
 
         uint256 nonce;
         unchecked {
             nonce = _nonces[request.revoker]++;
         }
 
         bytes32 digest = _hashTypedDataV4(keccak256(abi.encode(REVOKE_TYPEHASH, request.schema, data.uid, nonce)));
+        _verifySignature(digest, signature, request.revoker);
+    }
+
+    /// @notice Verifies EIP712 signatures (with EIP1271 support).
+    /// @param digest The typed-data digest to verify.
+    /// @param signature The signature to verify (either a "real" ECDSA signature or an EIP1271-aware signature).
+    /// @param signer The signer to verify the signature against.
+    function _verifySignature(bytes32 digest, Signature memory signature, address signer) private view {
```

Comment:
If we're going to be using openzeppelin's libraries, can we use their signature validation libraries as well as opposed to rolling our own?
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/9e3f4d60c581010c4a3979480e07cc7752f124cc/contracts/utils/cryptography/SignatureChecker.sol#L14

Reply:
cc @lbeder

Reply:
Yeah, we can use OZ's `SignatureChecker`, but it will require to pack the signature into `bytes`. Please let me know if this is preferred and I'll update it.

Reply:
I want to keep this in sync with your upstream as much as possible

Reply:
Sure thing. Let me address this in our upstream and then I'll open a PR to this PR 👍

Reply:
Ok thank you! Super appreciate it :D

Reply:
I've opened https://github.com/ethereum-optimism/optimism/pull/6750 (which was ported and tested from https://github.com/ethereum-attestation-service/eas-contracts/pull/81)

Reply:
Thank you so much!

Reply:
🤝